ngraph/core/reference/src/runtime/reference/convert.cpp

```diff
@@ -0,0 +1,273 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/reference/convert.hpp"
+#include "jit_generator.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            namespace
+            {
+                class jit_convert_u8_to_f16 : public jit::Generator
+                {
+                    jit_convert_u8_to_f16()
+                    {
+                        using namespace Xbyak;
+
+                        auto u8vec = xmm1;
+                        auto i32vec = ymm2;
+                        auto f16vec = xmm3;
+                        auto fvec = ymm4;
+
+                        auto reg_src = rax;
+                        auto reg_dst = rbx;
+                        auto reg_sz = rdx;
+
+                        Label loop, tail, tail_loop, copy_loop, exit;
+
+                        preamble();
+
+                        mov(reg_src, ptr[param + offsetof(args_t, arg)]);
+                        mov(reg_dst, ptr[param + offsetof(args_t, out)]);
+                        mov(reg_sz, ptr[param + offsetof(args_t, count)]);
+
+                        L(loop);
+
+                        cmp(reg_sz, 8);
```

Comment:
could you please use meaningful name for `8` constant. Commonly used is `vlen`.

Reply:
fixed